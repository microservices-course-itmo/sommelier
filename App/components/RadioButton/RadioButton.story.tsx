import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { boolean } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import { RadioButton } from './RadioButton'
import { BufferView, ThemeDecorator } from '../../config/decorators'

const defaultRadioButton = () => (
  <RadioButton
    checked={boolean('checked', true)}
    onPress={action('tapper-default')}
  />
)

storiesOf('RadioButton', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .add('Default RadioButton', defaultRadioButton)
