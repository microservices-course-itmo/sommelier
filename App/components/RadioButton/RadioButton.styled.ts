import styled from 'styled-components/native'

export const RadioButtonContainer = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  width: 100%;
`

export const RadioButtonText = styled.Text`
  color: black;
`

export const RadioButtonViewUnfilled = styled.View`
  height: 20px;
  width: 20px;
  border-radius: 10px;
  border-width: 1px;
  border-color: ${({ theme }) => theme.colors.fillGray};
  align-items: center;
  justify-content: center;
  margin-right: 40px;
`

export const RadioButtonViewFilled = styled.View`
  width: 20px;
  height: 20px;
  border-radius: 10px;
  border-color: ${({ theme }) => theme.colors.fillRed};
  background-color: ${({ theme }) => theme.colors.fillRed};
  align-items: center;
  justify-content: center;
  margin-right: 40px;
`
