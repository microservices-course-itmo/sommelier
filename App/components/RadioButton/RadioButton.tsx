import React, { ReactNode, useState } from 'react'
import {
  RadioButtonContainer,
  RadioButtonText,
  RadioButtonViewFilled,
  RadioButtonViewUnfilled,
} from './RadioButton.styled'

export type RadioButtonProps = {
  checked?: boolean
  children?: ReactNode
}

export const RadioButton = ({
  checked = true,
  children = '',
}: RadioButtonProps) => {
  const [pressed, setPressed] = useState(checked)
  return (
    <RadioButtonContainer onPress={() => setPressed(!pressed)}>
      {pressed ? <RadioButtonViewFilled /> : <RadioButtonViewUnfilled />}
      <RadioButtonText>{children}</RadioButtonText>
    </RadioButtonContainer>
  )
}
