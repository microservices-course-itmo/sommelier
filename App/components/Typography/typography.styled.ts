import styled from 'styled-components/native'
import {
  TypographyProps,
  StyledHeaderProps,
  StyledTextProps,
} from './typography.types'

const StyledTypography = styled.Text<TypographyProps>`
  color: ${({ color, theme }) =>
    color ? theme.colors[color] : theme.colors.black};
  font-weight: ${({ bold }) => (bold ? 'bold' : 'normal')};
  font-style: ${({ italic }) => (italic ? 'italic' : 'normal')};
  text-decoration: ${({ crossedOut }) =>
    crossedOut ? 'line-through' : 'none'};
`

export const StyledHeader = styled(StyledTypography)<StyledHeaderProps>`
  font-size: ${({ size, theme }) => theme.headerSizes[size]};
`

export const StyledText = styled(StyledTypography)<StyledTextProps>`
  font-size: ${({ size, theme }) => theme.fontSizes[size]};
`
