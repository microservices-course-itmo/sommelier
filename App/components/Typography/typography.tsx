import React, { ReactNode } from 'react'
import * as Styled from './typography.styled'
import theme from '../../theme/index'
import { ReactElement } from 'react'

type Props = {
  children: ReactNode
  color: string
  bold?: boolean
  italic: boolean
  size?: keyof typeof theme.fontSizes
  crossedOut?: boolean
  fontFamily: string
}

export const H1 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH1
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH1>
)

export const H2 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH2
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH2>
)

export const H3 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH3
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH3>
)

export const H4 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH4
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH4>
)

export const H5 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH5
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH5>
)

export const H6 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH6
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH6>
)

export const Text = ({
  children,
  color,
  bold,
  italic,
  size,
  crossedOut,
  fontFamily,
}: Props): ReactElement => (
  <Styled.StyledText
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
    size={size}
    crossedOut={crossedOut}
  >
    {children}
  </Styled.StyledText>
)
