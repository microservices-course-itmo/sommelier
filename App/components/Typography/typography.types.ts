import { ReactNode } from 'react'
import { COLOR_KEYS } from '../../theme/colors'
import { HEADER_SIZES_KEYS } from '../../theme/headerSizes'
import { FONT_SIZE_KEYS } from '../../theme/fontSizes'

type TypographyProps = {
  bold?: boolean
  italic?: boolean
  crossedOut?: boolean
  color?: keyof typeof COLOR_KEYS
}

export type StyledHeaderProps = TypographyProps & {
  size: keyof typeof HEADER_SIZES_KEYS
}
export type HeaderProps = StyledHeaderProps & { children: ReactNode }

export type StyledTextProps = TypographyProps & {
  size: keyof typeof FONT_SIZE_KEYS
}
export type TextProps = StyledTextProps & { children: ReactNode }
