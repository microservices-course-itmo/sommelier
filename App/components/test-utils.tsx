import React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const themeRender = (component: any) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
