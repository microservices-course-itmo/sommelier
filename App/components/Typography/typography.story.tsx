import React from 'react'
import styled from 'styled-components/native'
import { storiesOf } from '@storybook/react-native'
import { boolean, select } from '@storybook/addon-knobs'
import { H1, H2, H3, H4, H5, H6, Text } from './typography'
import { BufferView, ThemeDecorator } from '../../config/decorators'
import theme from '../../theme/index'

const StyledWrapper = styled.View`
  display: flex;
  flex-direction: column;
`

const FONT_SIZES = Object.keys(theme.fontSizes) as Array<
  keyof typeof theme.fontSizes
>
const COLORS = Object.keys(theme.colors) as Array<keyof typeof theme.colors>
const FONTS = Object.keys(theme.fonts) as Array<keyof typeof theme.fonts>

export const list = () => (
  <StyledWrapper>
    <H1 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H1 value
    </H1>
    <H2 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H2 value
    </H2>
    <H3 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H3 value
    </H3>
    <H4 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H4 value
    </H4>
    <H5 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H5 value
    </H5>
    <H6 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H6 value
    </H6>
    <Text
      color='black'
      bold={false}
      italic={false}
      fontFamily='PtSans'
      size='md'
    >
      Text value
    </Text>
  </StyledWrapper>
)

export const knobs = () => (
  <StyledWrapper>
    <H1
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H1 value
    </H1>
    <H2
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H2 value
    </H2>
    <H3
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H3 value
    </H3>
    <H4
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H4 value
    </H4>
    <H5
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H5 value
    </H5>
    <H6
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H6 value
    </H6>
    <Text
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
      size={select('text font-size', FONT_SIZES, FONT_SIZES[0])}
      crossedOut={boolean('cross-out', false)}
    >
      Text value
    </Text>
  </StyledWrapper>
)

storiesOf('Typography', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .add('Typography list', list)

storiesOf('Typography', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .add('Typography knobs', knobs)
