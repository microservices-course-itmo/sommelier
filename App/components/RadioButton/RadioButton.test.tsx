import React from 'react'
import { themeRender, fireEvent } from '../../config/test-utils'
import RadioButton from './RadioButton'
import { Text } from '../Typography'

describe('RadioButton component', () => {
  const mockFn = jest.fn()
  it('Component with only necessary props matches snapshot', () => {
    const tree = themeRender(<RadioButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Component with all props matches snapshot', () => {
    const tree = themeRender(
      <RadioButton checked={false}>
        <Text>abc</Text>
      </RadioButton>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Checked prop assigns correctly', () => {
    const elem = themeRender(<RadioButton checked={false} />)
    expect(elem.toJSON().children[1].props.checked).toBeFalsy()
  })
  // TODO: New functionality
  // it('onPress function should be called when pressing', () => {
  //   const elem = themeRender(<RadioButton onPress={mockFn} />)
  //   fireEvent.press()
  //   expect(mockFn).toHaveBeenCalled()
  // })
})
