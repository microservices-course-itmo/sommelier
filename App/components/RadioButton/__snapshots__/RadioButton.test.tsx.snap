// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component Component with all props matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#757575",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 20,
          "justifyContent": "center",
          "marginRight": 40,
          "width": 20,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "black",
        },
      ]
    }
  >
    <Text
      color="black"
      size="md"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          },
        ]
      }
      testID="sommelier-typography"
    >
      abc
    </Text>
  </Text>
</View>
`;

exports[`RadioButton component Component with only necessary props matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#A22841",
          "borderColor": "#A22841",
          "borderRadius": 10,
          "height": 20,
          "justifyContent": "center",
          "marginRight": 40,
          "width": 20,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "black",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
